src/enemyUtils.c

```diff
@@ -86,3 +92,139 @@ bool32 LoadEnemySprite(Entity* entity, const EnemyDefinition* definition) {
     LoadObjPalette(entity, definition->paletteIndex);
     return TRUE;
 }
+
+ASM_FUNC("asm/non_matching/enemyUtils/sub_0804A720.inc", void sub_0804A720(Entity* this))
+
+void CreateDeathFx(GenericEntity* parent, u32 parentId, u32 fixedItem);
+void sub_0804A7D4(Entity* this) {
+    CreateDeathFx((GenericEntity*)this, this->id, 0);
+}
+
+void CreateDeathFx(GenericEntity* parent, u32 parentId, u32 fixedItem) {
```

Comment:
Is this only called by enemies? If so it might make more sense to have a `GenericEnemy` struct instead

Reply:
Yes, but I think at the moment we lack the knowledge to properly make a `GenericEnemy`